parquet-common/src/main/java/org/apache/parquet/SemanticVersion.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Very basic semver parser, only pays attention to major, minor, and patch numbers.
+ * Attempts to do a little bit of validation that the version string is valid, but
+ * is not a full implementation of the semver spec.
+ *
+ * NOTE: compareTo only respects major, minor, and patch (ignores rc numbers, SNAPSHOT, etc)
```

Comment:
Minor: I don't think this would be too hard to fix, since the characters after patch indicate that the version comes before any version without an extra qualifier and the versions with qualifiers should be interpreted by a simple sort.